lib/chore/worker.rb

```diff
@@ -42,6 +42,24 @@ def expires_at
       @started_at + total_timeout
     end
 
+    def duplicate_work?(item)
+      # if we've got a duplicate, remove the message from the queue by not actually running and also not reporting any errors
+      payload = options[:payload_handler].payload(item.decoded_message)
+      
+      # only dedupe on the message id if we're getting work from an sqs consumer (currently the only consumer that can recieve the same message id twice)
+      return false if item.consumer.class != Chore::Queues::SQS::Consumer
+
+      # if the message is a duplicate on the queue id, re-enqueue it
```

Comment:
`re-enqueue it` isn't quite right here -- really we're assuming that some other worker has picked up the job and is processing it.  The job may or may not get re-enqueued depending on whether it succeeds on the other worker.